app/ssttparrangement/ArrangementController.scala

```diff
@@ -98,11 +98,11 @@ class ArrangementController @Inject() (
 
     for {
       tp: model.Taxpayer <- taxPayerConnector.getTaxPayer(asTaxpayersSaUtr(request.utr))
-      newJourney: Journey = Journey.newJourney.copy(maybeTaxpayer = Some(tp))
-      _ <- journeyService.saveJourney(newJourney)
-      result: Result <- eligibilityCheck(newJourney)
-    } yield result.placeInSession(newJourney._id)
-
+      maybeJourney <- journeyService.getMaybeJourney()
```

Comment:
please make sure the status is not finished

Reply:
Done

Reply:
Thanks!